next-cmake/CMakeLists.txt

```
@@ -39,7 +39,7 @@ include(hipblaslt_python)
 include(hipblaslt_target_configure_sanitizers)
 
 rocm_setup_version(VERSION ${PROJECT_VERSION})
-set(hipblaslt_SOVERSION 0.15)
+set(hipblaslt_SOVERSION 1.0)
```

```suggestion
set(hipblaslt_SOVERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR})
```
This sets the project version as the single source of truth as I don't see why the SO version and project version would be out of sync.

---

I think this depends on the used versioning scheme (which I am not aware of). If it is a semantic version number it indeed reflects the ABI version but if not, the ABI version and the version of the package might differ. No strong feelings though and for now I just ported what the commit applied to the "old" CMake config.

---

I merged as is to unblock your rocBLAS next-cmake efforts @bstefanuk, but we can iterate and change that later if needed.